core/controllers/contributor_dashboard.py

```diff
@@ -263,18 +264,47 @@ def _content_in_review(self, state_name, content_id, suggestions):
             s.change.content_id == content_id for s in suggestions)
 
 
-class MachineTranslatedStateTextsHandler(base.BaseHandler):
-    """Provide a machine translation of exploration content."""
+class MachineTranslationStateTextsHandler(base.BaseHandler):
+    """Provides a machine translation of exploration content."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
     @acl_decorators.open_access
     def get(self):
         """Handles GET requests. Responds with a mapping from content id to
         translation of form:
+
             dict('translated_texts', dict(str, str|None))
+
         If no translation is found for a given content id, that id is mapped to
         None.
+
+        Params:
+            exp_id: str. The ID of exploration being translated.
+            state_name: str. The name of the exploration state being translated.
+            content_ids: str[]. The content IDs of the texts to be translated.
+            target_language_code: str. The language code of the target
+                translation language.
+
+        Data Response:
+
+            dict(
+                'translated_texts', dict(str, str|None),
+                'errors': dict(str, str)
+            )
+
+            A dictionary containing the translated texts stored as a mapping
+                from content ID to the translated text. If an error occured
+                during retrieval of some content translations, but not others,
+                failed translations are mapped to None and the message is
+                returned in the 'errors' dict as a mapping from content id to
```

Comment:
I don't think you need to do this. Just return translated_texts with failed translations mapped to None, and do a logging.error() if there are any errors. The latter will be picked up in StackDriver and will be investigated. The user can't do anything about this anyway.

Reply:
Done.